### PR TITLE
Improve remove_nones to handle nested data structures

### PR DESCRIPTION
**Pull request by proelke**

### CODE REVIEW

File: ocpp/charge_point.py [hunk 1]
```diff
@@ -71,11 +71,13 @@ def snake_to_camel_case(data):
...
 def remove_nones(dict_to_scan):
-    dict_to_scan = {
-        k: v for k, v in dict_to_scan.items()
-        if v is not None
-    }
-    return dict_to_scan
+    new_dict = {}
+    for k, v in dict_to_scan.items():
+        if isinstance(v, dict):
+            v = remove_nones(v)
```
Comment (tropxy) [inline ocpp/charge_point.py:77]: what if the instance is a List?

If your payload is a Metervalue like this:
```python
MeterValuesPayload(connector_id=3, meter_value=[MeterValue(timestamp='2017-08-17T07:08:06.186748+00:00', sampled_value=[SampledValue(value='10', context='Sample.Periodic', format=None, measurand='Power.Active.Import', phase=None, location=None, unit='W'), SampledValue(value='50000', context='Sample.Periodic', format=None, measurand='Power.Active.Import', phase='L1', location=None, unit='W')]), MeterValue(timestamp='2017-08-17T07:07:07.186748+00:00', sampled_value=[SampledValue(value='10', context='Sample.Periodic', format=None, measurand='Power.Active.Import', phase=None, location=None, unit='W'), SampledValue(value='50000', context='Sample.Periodic', format=None, measurand='Power.Active.Import', phase='L1', location=None, unit='W')])], transaction_id=5)
```

which translates to a dict as:
```python
{'connector_id': 3,
 'meter_value': [{'timestamp': '2017-08-17T07:08:06.186748+00:00',
   'sampled_value': [{'value': '10',
     'context': 'Sample.Periodic',
     'format': None,
     'measurand': 'Power.Active.Import',
     'phase': None,
     'location': None,
     'unit': 'W'},
    {'value': '50000',
     'context': 'Sample.Periodic',
     'format': None,
     'measurand': 'Power.Active.Import',
     'phase': 'L1',
     'location': None,
     'unit': 'W'}]},
  {'timestamp': '2017-08-17T07:07:07.186748+00:00',
   'sampled_value': [{'value': '10',
     'context': 'Sample.Periodic',
     'format': None,
     'measurand': 'Power.Active.Import',
     'phase': None,
     'location': None,
     'unit': 'W'},
    {'value': '50000',
     'context': 'Sample.Periodic',
     'format': None,
     'measurand': 'Power.Active.Import',
     'phase': 'L1',
     'location': None,
     'unit': 'W'}]}],
 'transaction_id': 5}
```
And if you use the remove_nones, will yield:
```python
{'connector_id': 3,
 'meter_value': [{'timestamp': '2017-08-17T07:08:06.186748+00:00',
   'sampled_value': [{'value': '10',
     'context': 'Sample.Periodic',
     'format': None,
     'measurand': 'Power.Active.Import',
     'phase': None,
     'location': None,
     'unit': 'W'},
    {'value': '50000',
     'context': 'Sample.Periodic',
     'format': None,
     'measurand': 'Power.Active.Import',
     'phase': 'L1',
     'location': None,
     'unit': 'W'}]},
  {'timestamp': '2017-08-17T07:07:07.186748+00:00',
   'sampled_value': [{'value': '10',
     'context': 'Sample.Periodic',
     'format': None,
     'measurand': 'Power.Active.Import',
     'phase': None,
     'location': None,
     'unit': 'W'},
    {'value': '50000',
     'context': 'Sample.Periodic',
     'format': None,
     'measurand': 'Power.Active.Import',
     'phase': 'L1',
     'location': None,
     'unit': 'W'}]}],
 'transaction_id': 5}
```

I think the solution is:

```suggestion
        if isinstance(v, dict):
            v = remove_nones(v)
        if isinstance(v, list):
            new_list = []
            for item in v:
                new_list.append(remove_nones(item))
            v = new_list
```

**Review by tropxy — CHANGES_REQUESTED** [review]
Check it out

File: ocpp/charge_point.py (hunk 1 above)
Comment (proelke) [reply ocpp/charge_point.py:77]: Good call. I updated the code to handle this case and added a test.

Comment (tropxy) [reply ocpp/charge_point.py:77]: you welcome 😎

**Review by tropxy — APPROVED** [review]
lgtm

**Pull request merged**